### PR TITLE
DataSourcePicker: Design tweak proposal

### PR DESCRIPTION
**Pull request by torkelo**
Propose two changes:

* I think the current item design feels a bit messy with the ds type `| <Type>` suffix creating to much-unaligned text.  So propose we right align that text. And put the default tag to the right of name instead.
* Also the padding (1.5 grid units) and the ds logo image feel a bit too big. So reduced that padding to 1 and reduce the logo img from 32px to 24px. Happy to revert that. 

Before:
Screenshot from 2023-05-19 07-51-26

After: 
Screenshot 2023-05-22 at 14 28 02

Screenshot 2023-05-22 at 14 28 18

### CODE REVIEW

**Review by dprokop — APPROVED** [review]
Not sure about the logo size but the right-aligned text makes it way easier to scan the list for me.

**Comment by torkelo** [comment]
@dprokop yea, the logo size reduction was to be able to reduce the padding from 1.5 to 1 (grid units). Just made the sizing feel more natural to me. The 32px size logos feel a bit too overpowering and required a bit exceessive padding

File: packages/grafana-ui/src/components/Menu/SubMenu.tsx [hunk 1]
```diff
@@ -99,6 +99,7 @@ const getStyles = (theme: GrafanaTheme2) => {
       top: 0;
       left: 100%;
       z-index: ${theme.zIndex.dropdown};
+      border: 1px solid ${theme.colors.border.weak};
```
Comment (ivanortegaalba) [inline packages/grafana-ui/src/components/Menu/SubMenu.tsx:102]: Why this border? This is not used in DS dropdown

**Review by ivanortegaalba — COMMENTED** [review]
I'd not reduce the padding for the modal, or we should tweak the modal padding. The modal has a lot of blank space. 

Maybe we can only reduce the padding for the dropdown since the modal is big enough to display plenty of items and keep the current one for the modal.
I can imagine accounts with only 3-4 DS, and a big modal with a lot of blank space around.

Maybe, having a "compact" style for the dropdown, and the current one for the modal is better.

File: packages/grafana-ui/src/components/Menu/SubMenu.tsx (hunk 1 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/Menu/SubMenu.tsx:102]: unrelated change, will revert

**Review by ivanortegaalba — APPROVED** [review]
Ok! You're right @torkelo. The difference is small. LGTM.

**Pull request merged**